primer/src/Primer/App.hs

```diff
@@ -692,14 +692,39 @@ applyProgAction prog = \case
       ( m{moduleTypes = tydefs'}
       , Just $ SelectionTypeDef $ TypeDefSelection tc Nothing
       )
-  DeleteTypeDef d -> editModuleCross (qualifiedModule d) prog $ \(m, ms) ->
-    case moduleTypesQualified m Map.!? d of
-      Nothing -> throwError $ TypeDefNotFound d
-      Just (TypeDefPrim _) -> throwError $ TypeDefIsPrim d
-      Just (TypeDefAST td) -> do
-        checkTypeNotInUse d td $ m : ms
-        let m' = m{moduleTypes = Map.delete (baseName d) (moduleTypes m)}
-        pure (m' : ms, Nothing)
+  DeleteTypeDef d -> editModuleOfCrossType (Just d) prog $ \(m, ms) defName def -> do
+    let updateExpr = \case
+          Con _ c _ | c `elem` map valConName (astTypeDefConstructors def) -> emptyHole
+          e -> pure e
+    ms' <-
```

Comment:
It would be nice to replicate your comments from the PR into the code (here and elsewhere), otherwise future readers will probably lose that context (e.g. why not do anything about pattern matches?).